### PR TITLE
Enable display and deletion of schema templates

### PR DESCRIPTION
**Pull request by bigreybear**
### Content3 ...

<hr>

This PR has:
- [x] been self-reviewed.
    - [x] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Review by qiaojialin — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 1]
```diff
@@ -2233,6 +2250,137 @@ public boolean isPathExistsInTemplate(String templateName, String path) throws M
...
+              }
+
+              // descendants of this node may be using template too
+              if (node.isUseTemplate()) {
+                result.add(node.getFullPath());
+              }
+            }
+            return false;
+          }
+        };
```
Comment (zyk990424) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:2339]: encapsulate this in MTree

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 2]
```diff
@@ -2162,12 +2163,48 @@ public boolean isPathExistInTemplate(String templateName, String path)
...
+    return resp.getMeasurements();
+  }
+
   public void unsetSchemaTemplate(String prefixPath, String templateName)
       throws IoTDBConnectionException, StatementExecutionException {
     TSUnsetSchemaTemplateReq request = getTSUnsetSchemaTemplateReq(prefixPath, templateName);
     defaultSessionConnection.unsetSchemaTemplate(request);
   }
 
+  public void dropSchemaTemplate(String templateName)
```
Comment (SilverNarcissus) [inline session/src/main/java/org/apache/iotdb/session/Session.java:2202]: We should also update API doc at: docs/UserGuide/API/Programming-Java-Native-API.md

**Review by SilverNarcissus — APPROVED** [review]
Brilliant work!  only need to update doc~

File: session/src/main/java/org/apache/iotdb/session/Session.java (hunk 2 above)
Comment (bigreybear) [reply session/src/main/java/org/apache/iotdb/session/Session.java:2202]: Thanks, both english and chinese API docs are updated now

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 1 above)
Comment (bigreybear) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: It has been encapsulated now

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 3]
```diff
@@ -2233,6 +2250,137 @@ public boolean isPathExistsInTemplate(String templateName, String path) throws M
...
+              if (templateName.equals("")
+                  || templateName.equals(node.getUpperTemplate().getName())) {
+                resSet.add(node.getFullPath());
+              }
+              // descendants of the node cannot set another template, exit from this branch
+              return true;
+            }
+            return false;
+          }
+        };
```
[the remark this answers is not in the document]
Comment (bigreybear) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: It has been encapsulated now

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 4]
```diff
@@ -2233,6 +2246,47 @@ public boolean isPathExistsInTemplate(String templateName, String path) throws M
...
+      templateManager.dropSchemaTemplate(plan);
+      if (!isRecovering) {
+        logWriter.dropSchemaTemplate(plan);
+      }
+    } catch (IOException e) {
+      throw new MetadataException(e);
+    }
+  }
+
   public synchronized void setSchemaTemplate(SetTemplatePlan plan) throws MetadataException {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:2290]: Use MTree.setTemplate, do not get Node from MTree

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 5]
```diff
@@ -2233,6 +2246,47 @@ public boolean isPathExistsInTemplate(String templateName, String path) throws M
...
+        throw new UndefinedTemplateException(templateName);
+      }
+
+      if (mtree.isTemplateSetOnMTree(templateName)) {
+        throw new MetadataException(
+            String.format(
+                "Template [%s] has been set on MTree, cannot be dropped now.", templateName));
+      }
+
+      templateManager.dropSchemaTemplate(plan);
```
[the remark this answers is not in the document]
Comment (bigreybear) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:2281]: fixed

**Review by qiaojialin — APPROVED** [review]

**Review by zyk990424 — APPROVED** [review]
Nice feature! LGTM~

**Pull request merged**